kiegroup/kogito-cloud-operator
# KOGITO-2327: KogitoRuntime Support in Kubernetes

JIRA Ticket: https://issues.redhat.com/browse/KOGITO-2327
Description: KogitoRuntime Support in Kubernetes

Pending tasks:
- [x] Run basic test
- [x] Install Openshift community catalog in Kubernetes (I propose to do this as part of another ticket using a real Kubernetes instance)
- [x] Run tests with Infinispan and Kafka (Tests with Kafka are not working, but not because these changes)
- [x] Investigate metrics log errors when deploying the Kogito Operator (raised KOGITO-2404)

How to test locally:

1.- Run minikube (v1.11 or later):

```
minikube start --addons ingress --addons olm --driver kvm2
```

The OC client will be configured transparently for us.

2.- Configure your /etc/hosts file with:

```
[YOUR K8S IP] data-index
[YOUR K8S IP] process-springboot-example
[YOUR K8S IP] process-quarkus-example
```

To find the IP, run "minikube ip".

3.- Login to the quay.io repository using your account:

```
docker login quay.io
```

And make your repositories public:

```
process-springboot-example-default
process-quarkus-example-default
process-springboot-example-persistence
process-quarkus-example-persistence
process-springboot-example-events-persistence
process-quarkus-example-events-persistence
```

Note that these repositories are created in the next step. 

4.- Run the tests:

```
../hack/run-tests.sh --feature "features/deploy_kogito_runtime.feature" --runtime_application_image_namespace <quay account> --runtime_application_image_registry quay.io --cluster kubernetes --local-cluster
```

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
